server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -680,7 +718,7 @@ private boolean deleteIfEmpty() throws IOException {
     }
 
     public void close() throws IOException {
-        deleteStaleSegments(0);
```

Comment:
Not directly related to this change, but had few questions: 
- Shouldn't we invoke the `deleteStaleSegmentsAsync` only from primary shards (and not replicas)?
- Also, I wanted to understand the behaviour in case of primary relocation. Wouldn't we trigger `deleteStaleSegmentsAsync(0)` on the directory in such cases?

Reply:
> As per my understanding RemoteSegmentStoreDirectory doesn't understand shard, So invoking it just for primary will be tricky but will try to explore.

Understood. IMO, it does make sense for remote store directory to know if it is created by reader or writer.

Reply:
> These are anyways stale commit information even if the shards relocate, At some point we expect them to get deleted.

I wanted to understand if its possible that primary relocation leads to `deleteStaleSegments(0)` and thus deleting all the data in the remote store.

Reply:
Relocating shards will call `closeShards`  but `deleted.get()` will be false for these cases and deletion won't be triggered .